addons/stock/wizard/stock_lot_label_layout.py

```diff
@@ -25,12 +25,12 @@ def process(self):
         if self.label_quantity == 'lots':
             docids = self.move_line_ids.lot_id.ids
         else:
-            uom_categ_unit = self.env.ref('uom.product_uom_categ_unit')
+            uom_unit = self.env.ref('uom.product_uom_unit')
             quantity_by_lot = defaultdict(int)
             for move_line in self.move_line_ids:
                 if not move_line.lot_id:
                     continue
-                if move_line.product_uom_id.category_id == uom_categ_unit:
+                if uom_unit.parent_path in move_line.product_uom_id.parent_path:
```

Comment:
Now that I think about it, `uom_unit` will have a `parent_path` of `1/`. This may lead to a lot of false positives

Reply:
Indeed. I implemented a more robust method `has_common_reference` in `uom_uom` to be used for such checks: https://github.com/odoo/odoo/pull/184131/files#diff-7aa42e1663219ce82e34c884679a1f295e1fe26afd8daa7cb4d52d8663e79b83R118-R130.